src/swap/swapuz.ts

```diff
@@ -0,0 +1,308 @@
+            minAmount.toString(),
+            fromCurrencyCode
+          )
+          throw new SwapBelowLimitError(swapInfo, nativeMinAmount)
+        }
+
+        // Create order
+        const orderBody = {
+          from: fromCurrencyCode,
+          fromNetwork: fromMainnetCode,
+          to: toCurrencyCode,
+          toNetwork: toMainnetCode,
+          address: toAddress,
+          amount: parseFloat(largeDenomAmount),
+          mode,
+          addressUserFrom: fromAddress,
+          addressRefound: fromAddress
+        }
+
+        const createOrderResponse = await fetch(uri + 'order', {
+          method: 'POST',
+          body: JSON.stringify(orderBody),
+          headers
+        })
+        if (!createOrderResponse.ok) {
+          throw new Error(
+            `Swapuz call returned error code ${createOrderResponse.status}`
+          )
+        }
+
+        const createOrderJson = asApiResponse(asCreateOrder)(
+          await createOrderResponse.json()
+        )
+
+        if (createOrderJson.result == null) {
+          throw new SwapCurrencyError(
+            swapInfo,
+            fromCurrencyCode,
+            toCurrencyCode
+          )
+        }
+
+        const {
+          addressFrom,
+          finishPayment,
+          amountResult,
+          uid,
+          memoFrom
+        } = createOrderJson.result
+
+        const spendInfo: EdgeSpendInfo = {
+          currencyCode: fromCurrencyCode,
+          spendTargets: [
+            {
+              nativeAmount: request.nativeAmount,
+              publicAddress: addressFrom,
+              uniqueIdentifier: memoFrom
+            }
+          ],
+          networkFeeOption: fromCurrencyCode === 'BTC' ? 'high' : 'standard',
+          swapData: {
+            orderId: uid,
+            orderUri: orderUri + uid,
+            isEstimate: mode === 'float',
+            payoutAddress: toAddress,
+            payoutCurrencyCode: toCurrencyCode,
+            payoutNativeAmount: nativeAmount,
+            payoutWalletId: request.toWallet.id,
+            plugin: { ...swapInfo },
+            refundAddress: fromAddress
+          }
+        }
+
+        const tx: EdgeTransaction = await request.fromWallet.makeSpend(
+          spendInfo
+        )
+
+        const toNativeAmount = await toWallet.denominationToNative(
+          amountResult.toString(),
+          toCurrencyCode
+        )
+
+        return makeSwapPluginQuote(
+          request,
+          request.nativeAmount,
+          toNativeAmount,
+          tx,
+          toAddress,
+          pluginId,
+          mode === 'float',
+          ensureInFuture(finishPayment),
+          uid
+        )
+      }
+
+      // Try them all
+      try {
+        return await getQuote('fix')
+      } catch (e) {
+        try {
+          return await getQuote('float')
+        } catch (e2) {
+          // Should throw the fixed-rate error
+          throw e
+        }
+      }
+    }
```

Comment:
This is fine, but since these are all promises, I wonder if `return await getQuote('fix').catch(() => getQuote('float')).catch(e => e // should throw fixed-rate error)` would be prettier to look at, taking advantage of the promise chaining instead of the try / catch syntax.